src/parser/ParsedQuery.cpp

```diff
@@ -252,23 +252,33 @@ void ParsedQuery::expandPrefixes() {
   }
 
   vector<GraphPattern*> graphPatterns;
-  graphPatterns.push_back(_rootGraphPattern.get());
+  graphPatterns.push_back(&_rootGraphPattern);
   // Traverse the graph pattern tree using dfs expanding the prefixes in every
   // pattern.
   while (!graphPatterns.empty()) {
     GraphPattern* pattern = graphPatterns.back();
     graphPatterns.pop_back();
-    for (const std::shared_ptr<GraphPatternOperation>& p : pattern->_children) {
-      if (p->_type == GraphPatternOperation::Type::SUBQUERY) {
-        // Pass the prefixes to the subquery and expand them.
-        p->_subquery->_prefixes = _prefixes;
-        p->_subquery->expandPrefixes();
-      } else {
-        for (const std::shared_ptr<GraphPattern>& pattern :
-             p->_childGraphPatterns) {
-          graphPatterns.push_back(pattern.get());
+    for (GraphPatternOperation& p : pattern->_children) {
+      p.visit([&graphPatterns, this](auto&& arg) {
+        using T = std::decay_t<decltype(arg)>;
+        if constexpr (std::is_same_v<T, GraphPatternOperation::Subquery>) {
+          arg._subquery._prefixes = _prefixes;
+          arg._subquery.expandPrefixes();
+        } else if constexpr (std::is_same_v<T,
+                                            GraphPatternOperation::TransPath>) {
+          AD_CHECK(false);
+          // we may never be in an transitive path here or a
```

Comment:
The property paths are not yet expanded into trees of operations, thats done in the QueryPlanner. Due to that they are currently all contained in a triple, and the TransPath GraphPatternOperation cannot appear here. Instead, the prefix expansion is done as part of the prefix expansion of triples, a few lines below this one.